cocli/data/templates/comid-cca-realm-refval.json

```diff
@@ -0,0 +1,70 @@
+{
+    "lang": "en-GB",
+    "tag-identity": {
+        "id": "43BBE37F-2E61-4B33-AED3-53CFF1428B16",
+        "version": 0
+    },
+    "entities": [
+        {
+            "name": "Workload Client Ltd.",
+            "regid": "https://workloadclient.example",
+            "roles": [
+                "tagCreator",
+                "creator",
+                "maintainer"
+            ]
+        }
+    ],
+    "triples": {
+        "reference-values": [
+            {
+                "environment": {
+                    "class": {
+                        "id": {
+                            "type": "uuid",
+                            "value": "CD1F0E55-26F9-460D-B9D8-F7FDE171787C"
+                        },
+                        "vendor": "Workload Client Ltd"
+                    }
+                },
+                "measurements": [
+                    {
+                        "value": {
+                            "digests": [
+                                "sha-384;QoS1aUymwNLPR4mguVrIAlyBjeUjBDZL580pgbLS7caFsyInfsJYGZYkE9jJssH1"
+                            ]
+                        }
+                    },
+                    {
+                        "value": {
+                            "digests": [
+                                "sha-384;IQe752H8pS2VE2oTVNt6TdV7Gya+DT2nHZ6yOYazS6YVq/ZRTPNeWp6lWgMtBop4"
+                            ]
+                        }
+                    },
+                    {
+                        "value": {
+                            "digests": [
+                                "sha-384;QQe752H8pS2VE2oTVNt6TdV7Gya+DT2nHZ6yOYazS6YVq/ZRTPNeWp6lWgMtBop4"
+                            ]
+                        }
+                    },
+                    {
+                        "value": {
+                            "digests": [
+                                "sha-384;JQe752H8pS2VE2oTVNt6TdV7Gya+DT2nHZ6yOYazS6YVq/ZRTPNeWp6lWgMtBop4"
+                            ]
+                        }
+                    },
+                    {
+                        "value": {
+                            "digests": [
+                                "sha-384;MQe752H8pS2VE2oTVNt6TdV7Gya+DT2nHZ6yOYazS6YVq/ZRTPNeWp6lWgMtBop4"
+                            ]
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+}
```

Comment:
This looks very clumsy.  I suggest we extend the CoRIM measurement map with CCA-specific claims instead:

```json
[
  {
    "environment": {
      "instance": {
        "id": {
          "type": "arm-cca.rim",
          "value": "1000000000000000000000000000000000000000000000000000000000000000"
        }
      }
    }
  },
  {
    "measurement-map": {
      "mval": {
        "personalization": "5468652...",
        "RIM": "1000000000000000000000000000000000000000000000000000000000000000",
        "REM": [
          "0000000000000000000000000000000000000000000000000000000000000001",
          "0000000000000000000000000000000000000000000000000000000000000010",
          "0000000000000000000000000000000000000000000000000000000000000100",
          "0000000000000000000000000000000000000000000000000000000000010000"
        ],
        "RAK-hash-algo": "sha-256"
      }
    }
  }
]
```

Reply:
RAK Hash Algorithm Identifier and Personalization Values will NOT come from Supply Chain!
They will not be provisioned.

Only provisioned will be a Workload Identity, RIM and REM.

Reply:
Also effectively RIM and REM are Digests and using Digest to represent Reference Values is perfectly correct!

Reply:
> Also effectively RIM and REM are Digests and using Digest to represent Reference Values is perfectly correct!
Instead of minting custom `map` with no added gain!

Reply:
> RAK Hash Algorithm Identifier and Personalization Values will NOT come from Supply Chain!

why not?  And if not, should we expect another actor to supply those values?  If so, who?

Reply:
I am sure about RPV it is set internally by the Hypervisor and if I understand correctly, external parties will not have visibility of the values. For RAK Hash Algorithm, I will check and update this thread!

Reply:
> I am sure about RPV it is set internally by the Hypervisor and if I understand correctly, external parties will not have visibility of the values. 

no, both RPV and measurement algo are supplied as parameters at realm instantiation time to the VMM.

Reply:
They are supplied at Realm Instantiation time to RMM by host, If I understand correctly!

Reply:
yep, so it's not set internally.  It's externally provided.

Anyway, I don't see any problem with making those optional (alongside REMs).

Reply:
> I need to check again with our CCA experts before adding this as an Optional Parameter!
> 
> I am happy to do that, if I find that there is a possibility `Hypervisor` Gets the same from external sources:

```shell
 $ lkvm run --realm 				 \
	 --measurement-algo=["sha256", "sha512"] \
	 --disable-sve				 \
	 <normal-vm-options>
```

😄

Reply:
the measurement-algo here may be the hash algorithm used for Calculating RIM and REM Measurements? Isn't it?

Reply:
*all* realm measurmements (see §A7.2.3.1.5 of the realm spec.)

Reply:
yes this `measurement-algo` is expected to be the one set by External Supply Chain.

It is different then RAK Hash Algo, which was referred earlier!